packages/core/src/service/plugin/index.js

```diff
@@ -141,17 +141,31 @@ export default class PluginService {
   /**
    * Finds a layout plugin based on its name and version.
    */
-  findLayoutPlugin = (name: string, version: string): LayoutPlugin => {
+  findLayoutPlugin = (name: string, version: string): { plugin: LayoutPlugin, pluginWrongVersion?: LayoutPlugin } => {
     const plugin = this.plugins.layout.find(find(name, version))
-    return plugin || new LayoutPlugin(layoutMissing({ name, version }))
+    let pluginWrongVersion = undefined
+    if (!plugin) {
+      pluginWrongVersion = this.plugins.layout.find(find(name, '*'))
+    }
+    return {
+      plugin: plugin || new LayoutPlugin(layoutMissing({ name, version })),
+      pluginWrongVersion
+    }
   }
 
   /**
    * Finds a content plugin based on its name and version.
    */
-  findContentPlugin = (name: string, version: string): ContentPlugin => {
+  findContentPlugin = (name: string, version: string): { plugin: ContentPlugin, pluginWrongVersion?: ContentPlugin } => {
     const plugin = this.plugins.content.find(find(name, version))
-    return plugin || new ContentPlugin(contentMissing({ name, version }))
+    let pluginWrongVersion = undefined
+    if (!plugin) {
+      pluginWrongVersion = this.plugins.content.find(find(name, '*'))
+    }
+    return {
```

Comment:
So the API return value changed here, right? What's the the difference between `plugin` and `pluginWrongVersion`?

Reply:
plugin is the exact version, pluginWrongVersion is higher version of the plugin. So let's say you installed 0.0.2 and the data asks for 0.0.1. The plugin will be populated with the MissingPlugin fallback while the pluginWrongVersion will be populated with 0.0.2 version. You can then devide what to do. Yeah the api changed here but I think this is private api, right? at least I didn't find a place where it was exposed.

Reply:
Not sure if I understand the API correctly. I'd suggest to go with the idea raised in comment https://github.com/ory/editor/pull/554#discussion_r210720052 . It would remove the need for fallbacks and give plugin authors the ability to define version ranges themselves. If the version range for the installed plugin does not match the version from the content model, we show a plugin not found error, or a plugin does not match content version error.